plugins/google-sheets/src/main.tsx

```diff
@@ -21,23 +19,10 @@ export const queryClient = new QueryClient({
     },
 })
 
-function shouldSyncImmediately(pluginContext: PluginContext): pluginContext is PluginContextUpdate {
-    if (pluginContext.type !== "update") return false
-    if (pluginContext.slugFieldColumnIndex === null) return false
-    if (pluginContext.hasChangedFields) return false
-
-    return true
-}
-
 function renderPlugin(app: ReactNode) {
     const root = document.getElementById("root")
     if (!root) throw new Error("Root element not found")
 
-    framer.showUI({
```

Comment:
Moving this inside react components because synchronizing is now also done in the `<App />` component. The reason is this flow:

1. Press sync
2. You are not authenticated
3. You authenticate and we call `getContext` again
4. It now synchronizes without UI as expected.